include/revng/Support/IntegerSerialization.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/Support/Endian.h"
+
+#include "revng/ADT/CompilationTime.h"
+#include "revng/ADT/Concepts.h"
+
+namespace detail {
+
+inline constexpr auto IntEndianness = llvm::support::little;
+inline constexpr auto Unaligned = llvm::support::unaligned;
+
+template<SpecializationOf<std::tuple> T>
+inline constexpr size_t computeSize() {
+  size_t Result = 0;
+  compile_time::repeat<std::tuple_size_v<T>>([&Result]<size_t I>() {
+    Result += sizeof(std::tuple_element_t<I, T>);
```

Comment:
I think you're assuming you `tuple` is composed by primitive types, but you don't assert it anywhere.
What if it's another tuple?

You could recur instead of using `sizeof`. Otherwise, just assert.

Please document what is the scope of this library. What do you want to handle? How generic do you want to be?

Reply:
> but you don't assert it anywhere

It's asserted as part of the concept in `writeNext` and `readNext` on the individual elements

Reply:
Added another `static_assert` in `computeSize` that checks that the element type is one that we expect